algolia/opt/composable_filter_test.go

```diff
@@ -84,6 +84,20 @@ func TestComposableFilterOption_UnmarshalJSON(t *testing.T) {
 				{`color:blue`},
 			}},
 		},
+		{
+			`["color:green,color:yellow","color:blue"]`,
+			composableFilterOption{[][]string{
+				{`color:green,color:yellow`},
+				{`color:blue`},
+			}},
+		},
+		{
+			`"(color:green,color:yellow),color:black"`,
+			composableFilterOption{[][]string{
+				{`color:green`, `color:yellow`},
+				{`color:blue`},
```

Comment:
```suggestion
				{`color:black`},
```
Should we test the payload also?

Reply:
Let me check this out. I saw that other related tests(facet, tags etc...) are checking the payload as well but this one seems not. 
I am not sure if there was a reason but if it will be possible, I will try to add that to make tests stronger.

Reply:
It looks like the test was only comparing `len` of the slices. Removing that did the trick. 

https://github.com/algolia/algoliasearch-client-go/pull/718/commits/734da0f4f7c2e1b7c3fed70a318660b4eebc770b

Reply:
But `black` and `blue` are not of the same length so it should have failed :thinking:

Reply:
It wasn't checking the actual value length. It was checking slice length only. For example, the expected slice has a length of 2 and the actual slice has a length of 2. Please see the picture below, `fGot` and `fExpected` are `[][]string`. 

<img width="1703" alt="image" src="https://user-images.githubusercontent.com/33124154/221163467-2130816f-15e5-40b9-ad05-2301f5309962.png">


But with the change that I do now, it should check the whole slice with the elements. So it only passed after I changed `black` to `blue`.

Reply:
Thanks for the explanation, I should have re-read everything!

Reply:
no worries at all, thanks for following up 🙌